dags/iop/parser.py

```diff
@@ -0,0 +1,264 @@
+        raw_journal_doctype = article.find(
+            ".",
+        ).get("article-type")
+        journal_doctype = self.article_type_mapping[raw_journal_doctype]
+        if "other" in journal_doctype:
+            doi = article.find("front/article-meta/article-id/[@pub-id-type='doi']")
+            self.logger.msg(
+                "There are unmapped article types for article %s with types %s."
+                % (doi, raw_journal_doctype)
+            )
+        return journal_doctype
+
+    def _get_related_article_doi(self, article):
+        journal_doctype = self._get_journal_doctype(article)
+        # TODO: nee to find example with related_article_doi
+        if journal_doctype in ["correction", "addendum"]:
+            self.logger.msg("Adding related_article_doi.")
+            return article.find("./related-article[@ext-link-type='doi']/@href")
+
+    def _get_arxiv_eprints(self, article):
+        arxiv_eprints = []
+        arxivs_value = article.find(
+            "front/article-meta/custom-meta-group/custom-meta/meta-value"
+        ).text.lower()
+        pattern = re.compile(r"(arxiv:|v[0-9]$)", flags=re.I)
+        arxiv_eprints.append({"value": pattern.sub("", arxivs_value)})
+        return arxiv_eprints
+
+    def _get_institutions(self, article, reffered_id):
+        return article.findall(
+            f"front/article-meta/contrib-group/aff[@id='{reffered_id.get('rid')}']/institution"
+        )
+
+    def _get_countries(self, article, reffered_id):
+        return article.findall(
+            f"front/article-meta/contrib-group/aff[@id='{reffered_id.get('rid')}']/country"
+        )
+
+    def _get_affiliation_values(self, institutions, countries):
+        return [
+            {
+                "value": ",".join([institution.text, country.text]),
+                "country": country.text,
+            }
+            for institution, country in zip(institutions, countries)
+        ]
+
+    def _get_affiliations(self, article):
+        contrib_types = article.findall(
+            "front/article-meta/contrib-group/contrib[@contrib-type='author']"
+        )
+        authors = []
+        for contrib_type in contrib_types:
+            surname = contrib_type.find("name/surname").text
+            given_names = contrib_type.find("name/given-names").text
+            reffered_id = contrib_type.find("xref[@ref-type='aff']")
+            if reffered_id is None:
+                continue
+            institutions = self._get_institutions(article, reffered_id)
+            countries = self._get_countries(article, reffered_id)
+            affiliations = self._get_affiliation_values(institutions, countries)
+            author = {
+                "surname": surname,
+                "given_names": given_names,
+                "affiliations": affiliations,
+            }
+            authors.append(author)
+        return authors
+
+    # TODO: do I need it?
+
+    def _get_collab():
+        pass
```

Comment:
left over?

Reply:
no, I don't know do we need it. Anne said, "it might have collaborations."

Reply:
then we need an example and write code to parse it

Reply:
Asked Anne, to clarify about it

Reply:
let's completely remove this method